src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -280,9 +292,37 @@ def __new_installation_summary_json(self, installation_packages_json):
             "errors": self.__set_errors_json(self.__installation_total_error_count, self.__installation_errors)
         }
 
+    def set_patch_metadata_for_health_store_substatus_json(self, status=Constants.STATUS_SUCCESS, code=0, patch_version=Constants.PATCH_VERSION_UNKNOWN, report_to_health_store=False, wait_after_update=False):
+        """ Prepare the health store substatus json including message containing summary to be sent to health store """
+        self.composite_logger.log_debug("Setting patch metadata for health store substatus. [Substatus={0}] [Report to Health Store={1}]".format(str(status), str(report_to_health_store)))
+
+        # Wrap patch metadata into health store summary
+        self.__metadata_for_health_store_summary_json = self.__new_patch_metadata_for_health_store_json(patch_version, report_to_health_store)
+
+        # Wrap health store summary into health store substatus
+        self.__metadata_for_health_store_substatus_json = self.__new_substatus_json_for_operation(Constants.PATCH_METADATA_FOR_HEALTH_STORE, status, code, json.dumps(self.__metadata_for_health_store_summary_json))
+
+        # Update status on disk
+        self.__write_status_file()
+
+        # wait period required in cases where we need to ensure HealthStore reads the status from GA
+        if wait_after_update:
+            time.sleep(Constants.WAIT_FOR_AFTER_STATUS_UPDATE_IN_SECS)
+
+    def __new_patch_metadata_for_health_store_json(self, patch_version=Constants.PATCH_VERSION_UNKNOWN, report_to_health_store=False):
```

Comment:
healthstore is a single word here and elsewhere. 'health store' to 'healthstore', 'health_store' to 'healtstore' etc.